<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Sunburst</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.9.2/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.52.2/plotly-latest.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">OPERATING PROCEDURE R: SUNBURST VISUALIZATION OF THE CLASSIFICATION</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">INTRODUCTION</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    WALKTHROUGH
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="files.html">Files Needed</a>
    </li>
    <li>
      <a href="packages.html">Packages</a>
    </li>
    <li>
      <a href="path.html">Setting Path to Samples</a>
    </li>
    <li>
      <a href="colors.html">Colors</a>
    </li>
    <li>
      <a href="functions.html">Functions</a>
    </li>
    <li>
      <a href="sunburst.html">Sunburst Plot</a>
    </li>
  </ul>
</li>
<li>
  <a href="variablestochange.html">VARIABLES TO BE CHANGED</a>
</li>
<li>
  <a href="links.html">LINKS</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Sunburst</h1>

</div>


<p>Creating this sunburst will allow to more easily understand the taxonomic relationship between the different gastrointestinal species population found within our samples.</p>
<p>First, we need to use the <code>read.xlsx()</code> function to read specifically the IdTaxa60Classification sheet in the sequencing excel spreadsheet results. We will use the IdTaxa 60 classification results as it represents a confidence threshold of 60% which is higher than the IdTaxa 30 classification results, thus more robust. We also look at its structure to understand and look if changes need to be made.</p>
<pre class="r numberLines"><code>My_Data &lt;- read.xlsx(&quot;/Users/eleonorecharrier/Desktop/CODE/Summary.xlsx&quot;, sheet = &quot;IdTaxa60Classification&quot;)  #!!The first object of this function needs to be changed to wherever the `FinalSummary.xlxs` file is in your working directory</code></pre>
<p>After looking at the structure of our data frame “My_Data”, we need to create a new data frame containing only the column which will be of our interest in the creation of the Sunburst plot.</p>
<p>We will use the custom function “Function_Gather” create above. This allows us to only select the column <code>193_S193_L001_R1_001.fastq.gz</code>to the column<code>273_S273_L001_R1_001.fastq.gz</code> (includes all the fasta.gz column of the FinalSummary.xlxs file) and consider them as ‘Read’ values. This function gives us a new data frame “New_Data” grouped by taxonomy_value with the number of reads for each of these values.</p>
<pre class="r numberLines"><code>New_Data &lt;- Function_Gather(`193_S193_L001_R1_001.fastq.gz`:`273_S273_L001_R1_001.fastq.gz`, My_Data)</code></pre>
<p>We then need to define NA as a string character “Unknown” in the Taxonomy_Value column. This will be needed for the next step of rearranging the data is a custom order. Our data is a data frame thus, a named list giving the value, here “Unknown” is used to replace NA for each column using the <code>replace_na()</code> function.</p>
<pre class="r numberLines"><code>str(New_Data) #Check if the the colunm of interest were corectly selected </code></pre>
<pre><code>## Classes &#39;tbl_df&#39;, &#39;tbl&#39; and &#39;data.frame&#39;:    17 obs. of  2 variables:
##  $ Taxonomy_Value: chr  &quot;Chabertiidae&quot; &quot;Haemonchidae&quot; &quot;Haemonchus&quot; &quot;Haemonchus_placei&quot; ...
##  $ Total         : num  4409 230883 47433 42715 18953 ...</code></pre>
<pre class="r numberLines"><code>New_Data[17,1]=&quot;Unknown&quot;
New_Data %&gt;% replace_na(list(x = 0, y = &quot;Unknown&quot;))</code></pre>
<pre><code>## # A tibble: 17 x 2
##    Taxonomy_Value                Total
##    &lt;chr&gt;                         &lt;dbl&gt;
##  1 Chabertiidae                   4409
##  2 Haemonchidae                 230883
##  3 Haemonchus                    47433
##  4 Haemonchus_placei             42715
##  5 Molineidae                    18953
##  6 Nematodirus                   18953
##  7 Nematodirus_helvetianus       18953
##  8 Oesophagostomum                4409
##  9 Oesophagostomum_venulosum      4409
## 10 Ostertagia                    43716
## 11 Spiculopteragia              139734
## 12 Spiculopteragia_boehmi        48385
## 13 Strongylida                  254245
## 14 unclassified_Haemonchus        4718
## 15 unclassified_Ostertagia       43716
## 16 unclassified_Spiculopteragia  91349
## 17 Unknown                       19368</code></pre>
<p>This is the last step in rearranging our new data frame. We now have our data in two column, one for the taxonomy classification and one for the number of reads allocated to each level of the taxonomy classification. The last step is to arrange the data in the custom order needed for the organisation of the sunburst plot. To do so, we will use the custom function “Function_Arrange” create above.</p>
<pre class="r numberLines"><code>New_Data_Arrange &lt;- Function_Arrange(New_Data, New_Data$Taxonomy_Value ,c(&quot;Strongylida&quot;, &quot;Unknown&quot;, &quot;Haemonchidae&quot;, &quot;Chabertiidae&quot;, &quot;Molineidae&quot;, &quot;Spiculopteragia&quot;, &quot;Ostertagia&quot;, &quot;Haemonchus&quot;, &quot;Oesophagostomum&quot;, &quot;Nematodirus&quot;, &quot;Spiculopteragia_boehmi&quot;, &quot;unclassified_Spiculopteragia&quot;, &quot;unclassified_Ostertagia&quot;, &quot;Haemonchus_placei&quot;, &quot;unclassified_Haemonchus&quot;, &quot;Oesophagostomum_venulosum&quot;, &quot;Nematodirus_belvetianus&quot;))</code></pre>
<p>Using the custom function “Function_DataFrame_Sb” created above, we assign the ids, labels, parents and size to our sunburst plot.</p>
<pre class="r numberLines"><code>DF_Sb1 &lt;- Function_DataFrame_Sb(c(New_Data_Arrange$Taxonomy_Value),
          c(&quot;Strongylida&quot;, &quot;Unknown&quot;, &quot;Haemonchidae&quot;,  &quot;Chabertiidae&quot;, &quot;Molineidae&quot;, &quot;Spiculopteragia&quot;, &quot;Ostertagia&quot;, &quot;Haemonchus&quot;, &quot;Oesophagostomum&quot;, &quot;Nematodirus&quot;, &quot;Spiculopteragia&lt;br&gt;boehmi&quot;, &quot;unclassified&lt;br&gt;Spiculopteragia&quot;, &quot;unclassified&lt;br&gt;Ostertagia&quot;, &quot;Haemonchus&lt;br&gt;placei&quot;, &quot;unclassified&lt;br&gt;Haemonchus&quot;, &quot;Oesophagostomum&lt;br&gt;venulosum&quot;, &quot;Nematodirus&lt;br&gt;belvetianus&quot;),
          c(&quot;&quot;,&quot;&quot;, &quot;Strongylida&quot;,&quot;Strongylida&quot;,&quot;Strongylida&quot;, &quot;Haemonchidae&quot;, &quot;Haemonchidae&quot;, &quot;Haemonchidae&quot;,&quot;Chabertiidae&quot;, &quot;Molineidae&quot;, &quot;Spiculopteragia&quot;, &quot;Spiculopteragia&quot;, &quot;Ostertagia&quot;, &quot;Haemonchus&quot;, &quot;Haemonchus&quot;,&quot;Oesophagostomum&quot;, &quot;Nematodirus&quot;),
          New_Data_Arrange$Total)</code></pre>
<p>Finally, using the custom function “Function_Sb” created above we are constructing he sunburst plot of the “NUMBERS OF READS PER LEVEL OF TAXONOMY WITHIN DEER POPULATION”.</p>
<pre class="r numberLines"><code>Sb1 &lt;- Function_Sb(DF_Sb1,New_Data_Arrange$Total)
Sb1 ##To visualize our sunburst plot </code></pre>
<div id="htmlwidget-a603ba2ac83f38ddd7f5" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-a603ba2ac83f38ddd7f5">{"x":{"visdat":{"f6914fa66766":["function () ","plotlyVisDat"]},"cur_data":"f6914fa66766","attrs":{"f6914fa66766":{"ids":{},"labels":{},"parents":{},"hovertext":{},"marker":{"colors":["pink","black","palevioletred","#ef8a62","lightcoral","#a86c80","#a33358","#a85470","#fc8d62","#fbb4ae","#825564","#80666e","#c23e6a","#e9a3c9","#d47f9b","#d8b365","#ef8a62"]},"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"sunburst"}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"hovermode":"closest","showlegend":false},"source":"A","config":{"showSendToCloud":false},"data":[{"ids":["Strongylida","Unknown","Haemonchidae","Chabertiidae","Molineidae","Spiculopteragia","Ostertagia","Haemonchus","Oesophagostomum","Nematodirus","Spiculopteragia_boehmi","unclassified_Spiculopteragia","unclassified_Ostertagia","Haemonchus_placei","unclassified_Haemonchus","Oesophagostomum_venulosum","Nematodirus_helvetianus"],"labels":["Strongylida","Unknown","Haemonchidae","Chabertiidae","Molineidae","Spiculopteragia","Ostertagia","Haemonchus","Oesophagostomum","Nematodirus","Spiculopteragia<br>boehmi","unclassified<br>Spiculopteragia","unclassified<br>Ostertagia","Haemonchus<br>placei","unclassified<br>Haemonchus","Oesophagostomum<br>venulosum","Nematodirus<br>belvetianus"],"parents":["","","Strongylida","Strongylida","Strongylida","Haemonchidae","Haemonchidae","Haemonchidae","Chabertiidae","Molineidae","Spiculopteragia","Spiculopteragia","Ostertagia","Haemonchus","Haemonchus","Oesophagostomum","Nematodirus"],"hovertext":[254245,19368,230883,4409,18953,139734,43716,47433,4409,18953,48385,91349,43716,42715,4718,4409,18953],"marker":{"color":"rgba(31,119,180,1)","colors":["pink","black","palevioletred","#ef8a62","lightcoral","#a86c80","#a33358","#a85470","#fc8d62","#fbb4ae","#825564","#80666e","#c23e6a","#e9a3c9","#d47f9b","#d8b365","#ef8a62"],"line":{"color":"rgba(255,255,255,1)"}},"type":"sunburst","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>

<p>Copyright &copy; 2020, Gilleard Lab's, All rights reserved.</p>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
